source/common/http/header_utility.cc

```diff
@@ -239,7 +239,7 @@ bool HeaderUtility::isConnect(const RequestHeaderMap& headers) {
   return headers.Method() && headers.Method()->value() == Http::Headers::get().MethodValues.Connect;
 }
 
-bool HeaderUtility::isConnectUdp(const RequestHeaderMap& headers) {
+bool HeaderUtility::isConnectUdp(const RequestOrResponseHeaderMap& headers) {
```

Comment:
Why this change? Response headers don't indicate connect-udp

Reply:
Envoy will add the connect-udp upgrade header, see other comment for elaborate code explanation